File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
const reverseString = (str) => {
  // type your code here
    // input: Hello world
  // output: dlrow olleh
  // split the string into an array
  // iterate through the array backwards
  // push the letters into a new array
  // join the array into a string
  // return the string

  let splitString = str.split("");
  let reversedArray = [];
  for (let i = splitString.length - 1; i >= 0; i--) {
    reversedArray.push(splitString[i]);
  }
  return reversedArray.join("");
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
#input: hello world
#output: dlrow olleh
#split string into array
#reverse array
#join array into string
#return string

return str.split("").reverse.join("")

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution